### ISSUE TITLE
Customize a11y for TabBars

### ISSUE THREAD

**Issue by goderbauer** [flutter/flutter#12432]
Currently there is no way to customize what a TabBar exposes to a11y. For example, you can't modify the a11y label of a Tab. This is important because if your tab's title is "AD", VoiceOver will read it out as the individual letters "A" and "D" instead of the Word "Ad". 

To fix this, developers want to be able to provide a custom a11y label for a tab that's different from what's shown on screen.

In theory you should be able to do this with the help of the Semantics and ExcludeSemantics widgets. However, there is no way to insert that into a TabBar currently. You can't wrap the entire Tab in them because the TabBar only accepts children of a certain type. And you can't wrap just the text of the Tab in them because a Tab takes a string directly [1] and not a Text widget that could be wrapped in a Semantics widget.

[1] https://docs.flutter.io/flutter/material/Tab/text.html

**Comment by goderbauer** [comment]
The solution here is probably to add a child parameter to TabBar, that takes a Widget and if child is not provided, we are going to populate it with new Text(text).
